wordpress-varnish-as-a-service.php

```diff
@@ -9,17 +9,86 @@
 GitHub Plugin URI: https://github.com/UCF/UCF-WordPress-Varnish-as-a-Service
 Description: A plugin for purging Varnish cache when content is published or edited. It works with HTTP purge and Admin Port purge. Works with Varnish 2 (PURGE) and Varnish 3 (BAN) versions. Based on WordPress Varnish and Plugin Varnish Purges.
 */
+if ( ! defined( 'WPINC' ) ) {
+    die;
+}
+
 class WPVarnish {
 	public $commenter;
+	public $is_multisite=false;
+	public $is_network_activated=false;
+	public $plugin_group='wp-varnish-aas';
+
 	/**
 	 * __construct
 	 * Set default values and options for plugin.
 	 * Should be backward compatible back to PHP4.
 	 * @since v2.0.0
 	 **/
-	function __construct() {
+	public function __construct() {
 		global $post;
+		global $is_multisite;
+		global $is_network_activated;
 
+		if ( !function_exists( 'is_plugin_active_for_network' ) ) {
+			// Makes sure the plugin is defined before trying to use it
+			require_once ABSPATH . '/wp-admin/includes/plugin.php';
+		}
+		if ( is_multisite() ) {
+			$this->is_multisite=true;
+		}
+		else {
+			$this->is_multisite=false;
+		}
+		if ( is_multisite() && is_plugin_active_for_network( plugin_basename( __FILE__ ) ) ) {
+			$this->is_network_activated=true;
+			add_action('network_admin_menu', array(&$this, 'WPVarnishAdminMenu'));
+		}
+		else {
```

Comment:
This `else` block isn't really necessary, since $this->is_network_activated defaults to false (on line 19)